client/src/app/components/task-list/task-list/task-list.component.ts

```diff
@@ -55,16 +44,11 @@ export class TaskListComponent implements OnInit {
       confirmButtonText: 'Yes, delete it!',
     }).then(async (result) => {
       if (result.isConfirmed) {
-        (await this.taskService.deleteTask(id)).toPromise().then(() => {
-          Swal.fire({
-            title: 'Deleted!',
-            text: 'Item has been deleted.',
-            icon: 'success',
-          }).then(async (value) => {
-            if (value.isConfirmed) {
-              this.getTasksForMentee();
-            }
-          });
+        this.taskService.deleteTask(id).subscribe();
+        Swal.fire({
+          title: 'Deleted!',
+          text: 'Item has been deleted.',
+          icon: 'success',
         });
```

Comment:
You might want to put the `Swal.fire` inside the `next:` part of `subscribe()`? You only want to show the success message after the task was successfully deleted (and you could show an alert if there was an error)